packages/components/src/test/table/UsePresentationTable.test.tsx

```diff
@@ -171,4 +171,101 @@ describe("usePresentationTableWithUnifiedSelection", () => {
     expect(result.current.columns).to.have.lengthOf(0);
     expect(result.current.rows).to.have.lengthOf(0);
   });
+
+  it("adds passed keys to the unified selection with onSelect", async () => {
+    const keys = new KeySet([createTestECInstanceKey()]);
+    const stringifiedKeys: string[] = [];
+    const validKeys: Key[] = [];
+
+    keys.forEach((key) => {
+      stringifiedKeys.push(JSON.stringify(key));
+      validKeys.push(key);
+    });
```

Comment:
when testing valid situations, could we just take `rows` from `usePresentationTableWithUnifiedSelection` result and use their `key`? that would mimic the real world scenario.